README.md

```diff
@@ -3,21 +3,176 @@
+
+The Vault contracts are ERC4626 contracts used to manage the underlying ERC20 assets of the LP Providers. They are used to store the assets of the users and to allow them to deposit and withdraw assets that get lent to Market contracts for a fee which is agreed upon in the `setMarket` function. The users are minted a ERC20 share that represents their share of the underlying assets in the Vault.
+
+The following is a worked example of the relationship between users' deposits and shares.
+
+1. Alice deposits 1000 DAI into the Vault and receives 1000 shares.
+2. Bob deposits 1000 DAI into the Vault and receives 1000 shares.
+
+| User  | Action  | Amount   | Shares | Total Assets | Total Shares |
+| ----- | ------- | -------- | ------ | ------------ | ------------ |
+| Alice | Deposit | 1000 DAI | 1000   | 1000 DAI     | 1000         |
+| Bob   | Deposit | 1000 DAI | 1000   | 2000 DAI     | 2000         |
+
+The Vault is now holding 2000 DAI in "totalAssets". If Alice withdraws 500 shares, she will receive 500 DAI. Now, let's say a bet of 1800 DAI is placed on a market that is backed by the Vault at 1:1. The Vault will lend 1800 DAI to the market. 200 DAI remain in the vault and it will have a total exposure of 1800 DAI.
+
+3. Vault lends 1800 DAI to the market and has 200 DAI left in total assets.
+
+| User  | Action  | Amount   | Shares | Total Assets | Total Shares |
+| ----- | ------- | -------- | ------ | ------------ | ------------ |
+| Alice | Deposit | 1000 DAI | 1000   | 1000 DAI     | 1000         |
+| Bob   | Deposit | 1000 DAI | 1000   | 2000 DAI     | 2000         |
+| Vault | Lend    | 1800 DAI | 0      | 200 DAI      | 2000         |
+
+While the bet is active, Alice can redeem her 1000 shares if she chooses so, but will but the exchange rate per share will be significantly lower than 1 DAI per share. Her share is represented by the following equation:
+
+```text
+share = (totalAssets * shares) / totalShares
+```
+
+Eg:
+
+```text
+share = (200 * 1000) / 2000 = 100 DAI
+```
+
+Upon redeeming the shares will be burnt and Alice share balance will be 0. The Vault will have 1100 DAI in total assets and 1000 shares.
+
+4. Alice redeems 1000 shares for 100 DAI. Those shares are burnt.
+
+| User  | Action  | Amount   | Shares | Total Assets | Total Shares |
+| ----- | ------- | -------- | ------ | ------------ | ------------ |
+| Alice | Deposit | 1000 DAI | 1000   | 1000 DAI     | 1000         |
+| Bob   | Deposit | 1000 DAI | 1000   | 2000 DAI     | 2000         |
+| Vault | Lend    | 1800 DAI | 0      | 200 DAI      | 2000         |
+| Alice | Redeem  | 100 DAI  | (1000) | 100 DAI      | 1000         |
+
+The bet is now settled and the Vault has an exposure of 0 DAI and has made 1800 DAI on the losing bet. The total assets in the Vault are now 100 DAI. This would also hold true if the bet was a winning bet, and the market pays out the winning proposition.  The performance would be low as the bettor has one assets from the market.
```

Comment:
```suggestion
The bet is now settled and the Vault has an exposure of 0 DAI and has made 1800 DAI on the losing bet. The total assets in the Vault are now 2000 DAI. This would also hold true if the bet was a winning bet, and the market pays out the winning proposition.  The performance would be low as the bettor has one assets from the market.
```

Reply:
Actually, if they "made" 1800 DAI, wouldn't that mean that the market will transfer back a total of 3600 DAI?